add-interactive.c

```diff
@@ -7,6 +7,12 @@
 #include "revision.h"

```

Comment:
Could you leave an empty line between the two paragraphs in the commit message? Oh, and maybe "hence `add--helper --help` would be intercepted by `handle_builtin` and re-routed to the `help` command, without ever calling `cmd_add__helper()`"?

Reply:
Thanks! I was struggling with this one.